src/service.ts

```diff
@@ -55,175 +60,171 @@ interface ServiceOptions {
   token?: string;
 }
 
-/**
- * Service is a base class, meant to be inherited from by a "service," like
- * BigQuery or Storage.
- *
- * This handles making authenticated requests by exposing a `makeReq_` function.
- *
- * @constructor
- * @alias module:common/service
- *
- * @param {object} config - Configuration object.
- * @param {string} config.baseUrl - The base URL to make API requests to.
- * @param {string[]} config.scopes - The scopes required for the request.
- * @param {object=} options - [Configuration object](#/docs).
- */
-function Service(config: ServiceConfig, options?: ServiceOptions) {
-  options = options || {};
-
-  util.privatize(this, 'baseUrl', config.baseUrl);
```

Comment:
I'm not 100% what privatize does.  I tried making most of these fields private, but they ended up being used by the unit tests so I had to make them public.  My kingdom for `internal`. @ofrobots @stephenplusplus any ideas here?

Reply:
`privatize` was a new addition to resolve #13. In JS-land, it would just "hide" properties from `console.log`. I'm not sure what the modern equivalent is.

Reply:
Pretty sure that native class getters are not enumerable as well. Although I'm not sure if that's actually useful in this scenario.